File: seed/seed-portal/src/main/java/com/seed/portal/service/AppConfigService.java
package com.seed.portal.service;

import com.seed.base.model.PackVo;
import com.seed.data.model.so.AppConfigSo;
import com.seed.data.model.vo.AppConfigVo;

/**
 * AppConfigService
 *
 * @author <a href="mailto:shouheng2015@gmail.com">Shouheng.W</a>
 * @version 1.0 
 * @date 2020/11/06 08:36
 */
public interface AppConfigService {

    PackVo<AppConfigVo> createAppConfig(AppConfigVo vo);

    PackVo<AppConfigVo> getAppConfig(Long primaryKey);

    PackVo<AppConfigVo> updateAppConfig(AppConfigVo vo);

    PackVo<AppConfigVo> deleteAppConfig(Long primaryKey);

    PackVo<AppConfigVo> searchAppConfig(AppConfigSo so);

    PackVo<AppConfigVo> searchAppConfigCount(AppConfigSo so);

}

File: seed/APIFiles/AppConfigService.kt
package com.seed.data.api

/**
 * Get App Config
 *
 * @author <a href="mailto:shouheng2015@gmail.com">Shouheng.W</a>
 * @version 2020/11/07 02:39
 */
interface AppConfigService {

    /**
     * Api: Get App config
     * 
     * Desc: 
     * 
     * Parameter required: 
     * 1. apiParams.appId : current app id (Required): 
     * 2. apiParams.platform : current platform (Required): 
     * 3. apiParams.currentVersion : current version (Required): 
     * 4. apiParams.channel : current channel
     */
    @POST("/rest/config/getAppConfig")
    fun getAppConfigAsync(@Body request: BusinessRequest<AppConfigSo>): Deferred<BusinessResponse<AppConfigVo>>

}

